include/zephyr/bluetooth/mesh/msg.h

```diff
@@ -72,6 +72,22 @@ struct bt_mesh_model;
 #define BT_MESH_MODEL_BUF_DEFINE(_buf, _op, _payload_len)                      \
 	NET_BUF_SIMPLE_DEFINE(_buf, BT_MESH_MODEL_BUF_LEN(_op, (_payload_len)))
 
+/**
+ *  @brief Define a Mesh model message buffer using @ref NET_BUF_SIMPLE_DEFINE
+ *	   and initialization with opcode.
+ *
+ *  @param _buf         Buffer name.
+ *  @param _op          One octer opcode of the message.
+ *  @param _payload_len Length of the model message payload.
+ */
+#define BT_MESH_MODEL_BUF_INIT(_buf, _op, _payload_len)					 \
+	NET_BUF_SIMPLE_DEFINE(_buf, BT_MESH_MODEL_BUF_LEN(_op, (_payload_len)),		 \
+			      __COND_CODE((BT_MESH_MODEL_OP_LEN(_op) == 3),		 \
```

Comment:
Why not `COND_CODE_1` which is a public api?

Reply:
it will build error.

Reply:
like this:
zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:21: error: pasting "_XXXX" and "(" does not give a valid preprocessing token
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)

Reply:
I always ignore the difference between preprocessing and compilation, which makes me mistakenly think that opcodes of three byte lengths can be represented by a macro, but reality tells me that this is not possible. The preprocessing stage cannot do these things. Therefore, I had to redefine three types of Macro, but this is also a compromise choice, because the Mesh stack basically uses 2-octers opcode, so it is not difficult to distinguish.

Reply:
But compared to the call `bt_mesh_model_msg_init` with duplicate opcode which already used declared by `BT_MESH_MODEL_BUF_DEFINE`  in the Main branch, the current PR makes me look much more comfortable, haha.

Reply:
Ok, I see. But not, with `IS_EQ`, it will only work for opcodes < 4095: https://github.com/zephyrproject-rtos/zephyr/blob/a0ac2faf9bde45b176f1e8b315ffb1b8a9096166/include/zephyr/sys/util_internal.h#L52-L55

Reply:
Ah, sorry, this is opcode len!